### DISCUSSION TITLE
Kinda similar thing to Yuck

### DISCUSSION THREAD

**Discussion by njskalski** [elkowar/eww#866] [Show and tell]
Hi, for several years I work on https://njskalski.gitlab.io/triarii/bernardo/

I have a layout system that allows composing one widget from others.
I think I might re-use Yuck for it, but to do that, I'd need some docs, maybe tests.

Pros:
- yuck becomes more mature
Cons:
- we might have different design goals

What do you think?

**Comment by elkowar** [comment]
Sounds like a cool project!

To be honest, I'm not sure if re-using yuck is _worth it_:
The code is quite specific to eww, and is primarily related to interpreting yuck structures into eww-specific data structures. the core s-expression parser is quite trivial, as is the general nature of that sort of syntax. Lots of work also focusses on error handling, which again, is mostly specific to eww's usecase
